### PR TITLE
Purge Old DB Entries

### PR DESCRIPTION
**Pull request by yasin-cs-ko-ak**
• This PR contains generic config for purge old db entries for specific time.
• It uses cron job to run the db query.
• discovery-engine currently uses MySQL and SQLite as databases.
• `accuknox-obs.db` and `accuknox-pol.db` added in `local-file.yaml` as of now.

### CODE REVIEW

File: src/libs/mysqlHandler.go [hunk 1]
```diff
@@ -1724,3 +1725,24 @@ func GetSystemSummaryMySQL(cfg types.ConfigDB, filterOptions types.SystemSummary
...
+	defer db.Close()
+
+	timeNow := (ConvertStrToUnixTime("now"))
+	purgeTime := (config.GetCfgPublisherCronJobTime()) //sec
+	PurgeTimeValue, err := strconv.ParseInt(purgeTime, 10, 64)
+	if err != nil {
+		log.Error().Msg(err.Error())
+	}
+	ConvertedValue := timeNow - PurgeTimeValue
+	query := "DELETE FROM system_summary WHERE updated_time < " + strconv.Itoa(int(ConvertedValue))
```
Comment (rajaSahil) [inline src/libs/mysqlHandler.go:1744]: Shouldn't we add mutex lock while deleting the table?

File: src/libs/sqliteHandler.go [hunk 2]
```diff
@@ -1728,3 +1728,23 @@ func GetSystemSummarySQLite(cfg types.ConfigDB, filterOptions types.SystemSummar
...
+	defer db.Close()
+
+	timeNow := (ConvertStrToUnixTime("now"))
+	purgeTime := (config.GetCfgPublisherCronJobTime()) //sec
+	PurgeTimeValue, err := strconv.ParseInt(purgeTime, 10, 64)
+	if err != nil {
+		log.Error().Msg(err.Error())
+	}
+	ConvertedValue := timeNow - PurgeTimeValue
+	query := "DELETE FROM system_summary WHERE updated_time < " + strconv.Itoa(int(ConvertedValue))
```
Comment (rajaSahil) [inline src/libs/sqliteHandler.go:1746]: Same comment as above

File: src/libs/mysqlHandler.go (hunk 1 above)
Comment (yasin-cs-ko-ak) [reply src/libs/mysqlHandler.go:1744]: It is not necessary here. Because It's happening in specific time.

File: src/libs/sqliteHandler.go (hunk 2 above)
Comment (yasin-cs-ko-ak) [reply src/libs/sqliteHandler.go:1746]: It is not necessary here. Because It's happening in specific time.

File: src/libs/mysqlHandler.go (hunk 1 above)
Comment (rajaSahil) [reply src/libs/mysqlHandler.go:1744]: @yasin-cs-ko-ak Are we doing any other db operation for same table in any of the goroutines? 
cc: @seswarrajan

Comment (yasin-cs-ko-ak) [reply src/libs/mysqlHandler.go:1744]: @rajaSahil There is no need of goroutines. So I removed it. 
cc: @seswarrajan

File: src/libs/dbHandler.go [hunk 3]
```diff
@@ -616,3 +619,36 @@ func getSysSummarySQL(db *sql.DB, dbName string, filterOptions types.SystemSumma
...
+			return
+		}
+		PurgeDBCronJob.Start()
+		log.Info().Msg("Purging Old DB Entries cron job started")
+	}
+}
+
+func PurgeOldDBEntriesCronJob() {
+	if cfg.GetCfgPurgeOldDBEntriesEnable() {
+		PurgeOldDBEntriesMySQL(types.ConfigDB{})
```
Comment (seswarrajan) [inline src/libs/dbHandler.go:652]: Check the config choice of DB to use from conf/local-file.yaml and then call in a if else statement

Comment (seswarrajan) [reply src/libs/dbHandler.go:652]: if MySQL{
PurgeOldDBEntriesMySQL()
} else if SQLite {
PurgeOldDBEntriesSQLite()
}

**Review by seswarrajan — CHANGES_REQUESTED** [review]

File: src/libs/dbHandler.go (hunk 3 above)
Comment (yasin-cs-ko-ak) [reply src/libs/dbHandler.go:652]: @seswarrajan Changes are done.

File: src/libs/mysqlHandler.go (hunk 1 above)
Comment (seswarrajan) [reply src/libs/mysqlHandler.go:1744]: Yes @rajaSahil We will be doing add/update/read operations on this DB in another cron

**Review by seswarrajan — COMMENTED** [review]
Kindly respond to all comments by @rajaSahil

File: src/libs/mysqlHandler.go (hunk 1 above)
Comment (yasin-cs-ko-ak) [reply src/libs/mysqlHandler.go:1744]: As of now we are only purging the data. There is no other operations on this db other than that. @rajasahil

Comment (yasin-cs-ko-ak) [reply src/libs/mysqlHandler.go:1744]: As of now we are only purging the data. There is no other operations on this db other than that. @rajaSahil

**Comment by yasin-cs-ko-ak** [comment]
> Kindly respond to all comments by @rajaSahil

@seswarrajan All the comments are addressed. Thank you

**Pull request closed without merging**